### PR TITLE
[Admin REST API] - API to update block condition status

### PR DESCRIPTION
**Pull request by Meruja**
Fix https://github.com/wso2/product-apim/issues/8033 related to https://github.com/wso2/product-apim/issues/7954

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java [hunk 1]
```diff
@@ -979,6 +987,48 @@ public Response throttlingBlacklistConditionIdDelete(String conditionId, String
...
+                                                        String contentType, String ifMatch, String ifUnmodifiedSince,
+                                                        MessageContext messageContext) {
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            String username = RestApiUtil.getLoggedInUsername();
+
+            //This will give BlockConditionNotFoundException if there's no block condition exists with UUID
+            BlockConditionsDTO existingCondition = apiProvider.getBlockConditionByUUID(conditionId);
+            if (!RestApiAdminUtils.isBlockConditionAccessibleToUser(username, existingCondition)) {
+                RestApiUtil.handleAuthorizationFailure(RestApiConstants.RESOURCE_BLOCK_CONDITION, conditionId, log);
```
Comment (vithu30) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java:1011]: Don't we need to throw `404 - handleResourceNotFoundError` within else part to handle the case if the condition does not exist

Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java:1011]: 404 - handleResourceNotFoundError is handled in the catch block

**Comment by malinthaprasan** [comment]
Shall we have a proper title for this PR?

**Comment by Meruja** [comment]
> Shall we have a proper title for this PR?

The title has been revised.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java (hunk 1 above)
Comment (tgtshanika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java:1011]: If it is a non-existing block condition, does the method throw an APIM exception?

**Pull request merged**